api-compose-object.go

```diff
@@ -266,6 +266,51 @@ func (s *SourceInfo) getProps(c Client) (size int64, etag string, userMeta map[s
 	return
 }
 
+// Low level implementation of CopyObject API, supports only upto 5GiB worth of copy.
+func (c Client) copyObjectDo(ctx context.Context, srcBucket, srcObject, destBucket, destObject string,
+	metadata map[string]string) (ObjectInfo, error) {
+
+	// Build headers.
+	headers := make(http.Header)
+
+	// Set all the metadata headers.
+	for k, v := range metadata {
+		headers.Set(k, v)
+	}
```

Comment:
For metadata replacement to actually happen we need to make sure the `x-amz-metadata-directive` header is set to `REPLACE`. Perhaps we should throw an error if `metadata` is non-empty and this header is not set?

Reply:
You are right let me do that.

Reply:
actually we don't need to.. there is no error @donatello according to CopyObject() if REPLACE is set then the set metadata is honored, otherwise it does copy from the source object by default. So we don't need to error out.

Reply:
Ah yes. It is not needed, as AWS behaviour is also to silently ignore the headers without the directive header.